plugin/evm/customheader/gas_limit_test.go

```diff
@@ -9,27 +9,27 @@ import (
 
 	"github.com/ava-labs/avalanchego/utils/math"
 	"github.com/ava-labs/avalanchego/vms/components/gas"
+	"github.com/ava-labs/avalanchego/vms/evm/acp176"
 	"github.com/ava-labs/libevm/common"
 	"github.com/ava-labs/libevm/core/types"
 	"github.com/stretchr/testify/require"
 
 	"github.com/ava-labs/coreth/params/extras"
 	"github.com/ava-labs/coreth/plugin/evm/customtypes"
-	"github.com/ava-labs/coreth/plugin/evm/upgrade/acp176"
 	"github.com/ava-labs/coreth/plugin/evm/upgrade/ap0"
 	"github.com/ava-labs/coreth/plugin/evm/upgrade/ap1"
 	"github.com/ava-labs/coreth/plugin/evm/upgrade/ap5"
 	"github.com/ava-labs/coreth/plugin/evm/upgrade/cortina"
+	"github.com/ava-labs/coreth/utils"
 )
 
 func TestGasLimit(t *testing.T) {
```

Comment:
I see some good opportunities here to add the following test cases in `TestGasLimit`:

- `granite_initial_max_capacity` - Granite GasLimit at t=0 ms: Ensures GasLimit returns MaxCapacity on Granite genesis, validating the ms-based path and that GasLimit uses capacity ceiling, not current fill.

- `granite_after_1_5s_max_capacity` - Granite GasLimit at t=1500 ms: Confirms GasLimit remains MaxCapacity even mid-second, guarding against accidental dependence on fractional elapsed time.

Of course you would need to parameterize the `timestamp` and and set it this way when calling GasLimit:
```
got, err := GasLimit(config, test.parent, test.timestamp)
```